mol/molecule.py

```diff
@@ -3373,12 +3386,17 @@ def add_hatoms(self) -> "Molecule":
                     f_out.close()
                     f_out_bonds.close()
                     amsprep = os.path.join(os.environ["AMSBIN"], "amsprep")
+                    command = f"sh {amsprep} -t SP -m {f_in.name} -addhatoms -exportcoordinates {f_out.name} -bondsonly > {f_out_bonds.name}"
                     p = Popen(
-                        f"sh {amsprep} -t SP -m {f_in.name} -addhatoms -exportcoordinates {f_out.name} -bondsonly > {f_out_bonds.name}",
+                        command,
                         shell=True,
                         stdout=DEVNULL,
+                        stderr=PIPE,  # Redirect stderr to a pipe
                     )
-                    p.communicate()
+                    stdout, stderr = p.communicate()
+                    stderr_str = stderr.decode("utf-8").strip() if stderr else ""
+                    if stderr_str != "":
+                        print(f"amsprep raised: {stderr} \n run the command ${command} to get more info")
```

Comment:
```suggestion
                    _, stderr = p.communicate()
                    if stderr:
                        stderr_str = stderr.decode("utf-8").strip()
                        log(f"amsprep raised: {stderr_str} \n Run the command ${command} to get more information")
```
would use the plams log method here in place of the `print`

Reply:
to use the plams log is a good idea! 
And thanks for the correction about the `stderr`.When no error is raised `stderr==b''`. I did not know that `if b''` is equivalent to `if False`